driver-pulsar/src/main/java/io/nosqlbench/driver/pulsar/ops/PulsarConsumerOp.java

```diff
@@ -2,33 +2,49 @@
 
 import io.nosqlbench.driver.pulsar.util.AvroUtil;
 import io.nosqlbench.driver.pulsar.util.PulsarActivityUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.pulsar.client.api.*;
 import org.apache.pulsar.common.schema.SchemaType;
 
+import java.util.concurrent.TimeUnit;
+
 public class PulsarConsumerOp implements PulsarOp {
+
+    private final static Logger logger = LogManager.getLogger(PulsarConsumerOp.class);
+
     private final Consumer<?> consumer;
     private final Schema<?> pulsarSchema;
     private final boolean asyncPulsarOp;
+    private final int timeoutSeconds;
 
-    public PulsarConsumerOp(Consumer<?> consumer, Schema<?> schema, boolean asyncPulsarOp) {
+    public PulsarConsumerOp(Consumer<?> consumer, Schema<?> schema, boolean asyncPulsarOp, int timeoutSeconds) {
         this.consumer = consumer;
         this.pulsarSchema = schema;
         this.asyncPulsarOp = asyncPulsarOp;
+        this.timeoutSeconds = timeoutSeconds;
     }
 
     public void syncConsume() {
         try {
-            Message<?> message = consumer.receive();
+            Message<?> message = consumer.receive(timeoutSeconds, TimeUnit.SECONDS);
+            if (message == null) {
+                throw new RuntimeException("Did not receive a message within "+timeoutSeconds+" seconds");
+            }
 
             SchemaType schemaType = pulsarSchema.getSchemaInfo().getType();
             if (PulsarActivityUtil.isAvroSchemaTypeStr(schemaType.name())) {
-                String avroDefStr = pulsarSchema.getSchemaInfo().getSchemaDefinition();
-                org.apache.avro.generic.GenericRecord avroGenericRecord =
-                    AvroUtil.GetGenericRecord_ApacheAvro(avroDefStr, message.getData());
+                if (logger.isDebugEnabled()) {
```

Comment:
It would be better to provide a lambda-style logger line here, which we can do now that all the logging in NB is using log4j2. 
example:
            logger.debug(() -> {
                String avroDefStr = pulsarSchema.getSchemaInfo().getSchemaDefinition();
                org.apache.avro.generic.GenericRecord avroGenericRecord =
                    AvroUtil.GetGenericRecord_ApacheAvro(avroDefStr, message.getData());
                return "msg-key=" + message.getKey() + " msg-payload=" + avroGenericRecord;
            });

This is a matter of style preference, sure, but I think it makes the code easier to follow.